blaze-core/src/main/scala/org/http4s/blazecore/util/CachingChunkWriter.scala

```diff
@@ -42,37 +42,44 @@ private[http4s] class CachingChunkWriter[F[_]](
   import ChunkWriter._
 
   private[this] var pendingHeaders: StringWriter = _
-  private[this] var bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
+  private[this] val bodyBuffer: Buffer[Chunk[Byte]] = Buffer()
   private[this] var size: Int = 0
 
   override def writeHeaders(headerWriter: StringWriter): Future[Unit] = {
     pendingHeaders = headerWriter
     FutureUnit
   }
 
-  private def addChunk(chunk: Chunk[Byte]): Unit = {
-    bodyBuffer += chunk
-    size += chunk.size
-  }
-
-  private def clear(): Unit = {
+  private def addChunk(chunk: Chunk[Byte]): Unit =
+    if (chunk.nonEmpty) {
+      bodyBuffer += chunk
+      size += chunk.size
+    }
+
+  private def toChunkAndClear: Chunk[Byte] = {
+    val chunk = if (bodyBuffer.isEmpty) {
```

Comment:
This micro-opt seems legit - we won't add empty chunks to a buffer:
```suggestion
    val chunk = if (size == 0) {
```

Reply:
Actually, I think this will save a few nanoseconds, so If anyone else sees benefits from this micro opt - I'll do a PR.

Reply:
I think it's worth doing, as this is called a lot.